File: src/main/java/mygame/editor/actions/MoverAction.java
package mygame.editor.actions;


import com.badlogic.gdx.physics.box2d.BodyDef;
import javafx.geometry.Point2D;
import mygame.editor.model.Point;
import mygame.editor.render.CanvasRenderer;
import mygame.editor.repository.NodeRepository;
import mygame.editor.views.CcBodyNode;

/**
 * Created by oleh on 3/21/17.
 */
public class MoverAction extends Action implements CanvasRenderer.OnCanvasDragListener{


    public MoverAction(CanvasRenderer renderer, NodeRepository repository) {
        super(renderer,repository);
    }

    @Override
    public void init() {
        mRenderer.setOnCanvasDragListener(this);
    }


    @Override
    public void finishDrawing() {
        mRenderer.setOnCanvasDragListener(null);
    }

    @Override
    public void onStartMove(Point2D point) {

    }

    @Override
    public void onDrag(Point2D point) {

    }

    @Override
    public void onStopMove(Point2D point) {

    }
}

File: src/main/java/mygame/editor/actions/MoverAction.kt
package mygame.editor.actions;

import mygame.editor.model.PhysicsNode
import mygame.editor.ui.CustomPane;
import mygame.editor.view.AbstractView;
import mygame.editor.model.Point;
import mygame.editor.util.DragNDrop


/**
 * Created by oleh on 3/21/17.
 */
class MoverAction(parent: CustomPane, models: List<AbstractView>) : Action(parent, models) {

    override fun init() {

        for(child in parent.root.children){
            val listener = DragNDrop()
            child.onMouseDragged = listener.onMoveListener
            child.onMousePressed= listener.onStartListener

        }
    }

    override fun mouseMoved(position: Point) {

    }

    override fun mousePressed(position: Point) {

    }

    override fun mouseReleased(position: Point) {

    }

    override fun finishDrawing() {

    }
}
